test/wallet-rpc-test.js

```diff
@@ -313,4 +314,102 @@ describe('Wallet RPC Methods', function() {
       assert.strictEqual(info.height, node.chain.height);
     });
   });
+
+  describe('signmessagewithname & verifymessagewithname', () => {
+    const name = rules.grindName(5, 1, network);
+    const nonWalletName = rules.grindName(5, 1, network);
+    const message = 'Decentralized naming and certificate authority';
+
+    assert(name !== nonWalletName);
+
+    async function mineBlocks(n, addr) {
+      addr = addr ? addr : new Address().toString('regtest');
+      for (let i = 0; i < n; i++) {
+        const block = await node.miner.mineBlock(null, addr);
+        await node.chain.add(block);
+      }
+    }
+
+    before(async () => {
+      // Create a new wallets
+      await wclient.createWallet('alice');
+      const {receiveAddress} = await wclient.getAccount('alice', 'default');
+      await wclient.execute('selectwallet', ['alice']);
+
+      // fund wallet
+      await mineBlocks(2, receiveAddress);
+
+      // Win a name
+      await wclient.execute('sendopen', [name]);
+      await mineBlocks(network.names.treeInterval + 1);
+
+      await wclient.execute('sendbid', [name, 1, 2]);
+      await mineBlocks(network.names.biddingPeriod);
+
+      await wclient.execute('sendreveal', [name]);
+      await mineBlocks(network.names.revealPeriod + 1);
+    });
+
+    it('should sign and verify message with name', async () => {
+      await wclient.execute('selectwallet', ['alice']);
+
+      const signature = await wclient.execute('signmessagewithname', [
+        name,
+        message
+      ]);
+
+      const verify = await nclient.execute('verifymessagewithname', [
+        name,
+        signature,
+        message
+      ]);
+
+      assert.strictEqual(verify, true);
+    });
+
+    it('should fail with non-wallet name.', async () => {
+      await wclient.execute('selectwallet', ['alice']);
+
+      await assert.rejects(async () => {
+        await wclient.execute('signmessagewithname', [
+          nonWalletName,
+          message
+        ]);
+      }, {
+        type: 'RPCError',
+        message: 'Cannot find the name owner.'
+      });
+    });
+
+    ['', null, '\'null\'', 'localhost'].forEach((invalidName) => {
```

Comment:
Since you're not trying to include each `invalidName` in the title of the test, I think you can move the array and the loop (`for const...`) to the inside of the test block. You can also set the `const invalidNames = [...]` at the top of the `describe` section so you can just grab it from both tests.